README.adoc

```diff
@@ -122,16 +122,25 @@ QueryResource.java
 ----
 
-The `CompletionStage` interface represents a unit of computation.
-After a computation is complete, it can be either finished or chained with more `CompletionStage` interfaces using the [hotspot=thenAcceptAsync file=0]`thenAcceptAsync()` method. 
-Exceptions are handled in a callback provided to the [hotspot=exceptionally file=0]`exceptionally()` method, which behaves like a catch block.
-When you return a `CompletionStage` type in the resource, it doesn’t necessarily mean that the computation completed and the response was built.
-JAX-RS responds to the caller after the `CompletionStage` completes.
-
-A [hotspot=countdown1 hotspot=countdown2 hotspot=countdown3 file=0]`CountDownLatch` is used to track how many asynchronous requests that you're waiting on.
-By using the [hotspot=await file=0]`await` method, the program stops and waits until all our requests are complete. 
+The [hotspot=systemLoad file=0]`systemLoad` endpoint first gets all the hostnames by calling [hotspot=getSystems file=0]`getSystems()`.
+In the [hotspot=getSystem file=0]`getSystem()` method, multiple requests are sent to the `inventory` service for each hostname asynchronously. 
+When the requests return, the [hotspot=thenAcceptAsync file=0]`thenAcceptAsync()` method processes the returned data with the `CompletionStage<Properties>` interface.
 
-Declaring the `values` variable with the [hotspot=volatile file=0]`volatile` keyword and instantiating it as a [hotspot=concurrentHashMap file=0]`ConcurrentHashMap` object allows it to store system information and safely access that information asynchronously from multiple threads.
+The `CompletionStage<Properties>` interface represents a unit of computation.
+After a computation is complete, it can be either finished or chained with more `CompletionStage<Properties>` interfaces using the [hotspot=thenAcceptAsync file=0]`thenAcceptAsync()` method. 
+Exceptions are handled in a callback provided to the [hotspot=exceptionally file=0]`exceptionally()` method, which behaves like a catch block.
+When you return a `CompletionStage<Properties>` type in the resource, it doesn’t necessarily mean that the computation completed and the response was built.
+JAX-RS responds to the caller after the computation completes.
+
+In the [hotspot=systemLoad file=0]`systemLoad()` method a [hotspot=countdown1 hotspot=countdown2 hotspot=countdown3 file=0]`CountDownLatch` is used to keep track of the asynchronous requests. 
+The [hotspot=countdown2 hotspot=countdown3 file=0]`CountDownLatch` counts down whenever a request is complete.
+The `CountDownLatch` being at zero indicates that all asynchronous requests are complete.
+By using the [hotspot=await file=0]`await()` method of the `CountDownLatch`, the program waits for all the asynchronous requests to be complete.
+When all asynchronous requests are complete, the program will resume execution with all required data processed. 
```

Comment:
Suggested update:
"In the `systemLoad()` method, a `CountDownLatch` object is used to track asynchronous requests. 
The `CountDownLatch` counts down whenever a request completes.
When the `CountDownLatch` is at zero, it indicates that all asynchronous requests are complete.
By using the `await()` method of the `CountDownLatch`, the program waits for all the asynchronous requests to complete.
When all asynchronous requests are complete, the program resumes execution with all required data processed. "